test/functional/rpc_generateblock.py

```diff
@@ -16,72 +17,66 @@ class GenerateBlockTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
 
-    def skip_test_if_missing_module(self):
-        self.skip_if_no_wallet()
-
     def run_test(self):
         node = self.nodes[0]
+        miniwallet = MiniWallet(node)
+        miniwallet.rescan_utxos()
 
         self.log.info('Generate an empty block to address')
-        address = node.getnewaddress()
-        hash = self.generateblock(node, output=address, transactions=[])['hash']
+        address = miniwallet.get_address()
+        hash = node.generateblock(output=address, transactions=[], invalid_call=False)['hash']
```

Comment:
This is an invalid call. Can you explain why you are changing this line in the first place?

Reply:
ah okay I didn't appreciate the difference I had made. I've changed the calls back to how they were before.